### PR TITLE
feat(android): 🌟 add support for React Native 0.73

### PR DESCRIPTION
**Pull request by peterlazar1993**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :bulb: Motivation and Context
Starting from React Native v0.73 , all libraries will need to be updated with namespace due to the upgrade to AGP 8
https://github.com/react-native-community/discussions-and-proposals/issues/671

OS | Implemented
-- | --
iOS | ❌
Android | ✅

## :green_heart: How did you test it?
I tested the changes on RN 0.71.11 and everything works!

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [ ] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by fabiendem** [comment]
@krystofwoldrich 
> [...] Thank you for the PR. Before we merge, we need to make sure this is compatible with our min-supported RN version 0.65.3.

Do you folks really need to support RN down to 0.65.3?
RN is no longer maintained below 0.69.x:
https://github.com/reactwg/react-native-releases#releases-support-policy

**Comment by krystofwoldrich** [comment]
@fabiendem For the `v5` we have to and also want to respect `semver`. In case it's not possible to keep the SDK compatible we might think about bumping the version to v6.

**Comment by marandaneto** [comment]
You can make that compatible by doing this https://github.com/getsentry/sentry-dart/blob/ddc97adb5918b9f9eda573656e6edd9401b17801/flutter/android/build.gradle#L27-L30

**Comment by krystofwoldrich** [comment]
@marandaneto Awesome, thank you for sharing that.

File: android/build.gradle [hunk 1]
```diff
@@ -12,6 +12,7 @@ if (isNewArchitectureEnabled()) {
 }
 
 android {
+    namespace = "io.sentry.react"
```
Comment (krystofwoldrich) [inline android/build.gradle:15]: Lets first check if the namespace property is available not to break older apps with AGP<4.2.
```suggestion
    // Conditional for compatibility with AGP <4.2.
    if (project.android.hasProperty("namespace")) {
        namespace 'io.sentry.react'
    }
```

File: android/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,8 +1,8 @@
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="io.sentry.react">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <application>
-        <meta-data android:name="io.sentry.auto-init" android:value="false" />
+        <meta-data android:name="io.sentry.auto-init"
+            android:value="false" />
```
Comment (krystofwoldrich) [inline android/src/main/AndroidManifest.xml:6]:
```suggestion
        <meta-data android:name="io.sentry.auto-init" android:value="false" />
```

File: android/build.gradle [hunk 3]
```diff
@@ -12,7 +12,10 @@ if (isNewArchitectureEnabled()) {
 }
 
 android {
-    namespace = "io.sentry.react"
+    // Conditional for compatibility with AGP <4.2.
+    if (project.android.hasProperty("namespace")) {
+        namespace 'io.sentry.react'
```
Comment (krystofwoldrich) [inline android/build.gradle:17]:
```suggestion
        namespace = 'io.sentry.react'
```

File: android/build.gradle [hunk 4]
```diff
@@ -12,6 +12,10 @@ if (isNewArchitectureEnabled()) {
 }
 
 android {
+    // Conditional for compatibility with AGP <4.2.
+    if (project.android.hasProperty("namespace")) {
+        namespace = 'io.sentry.react'
```
Comment (krystofwoldrich) [inline android/build.gradle:17]:
```
        namespace = "io.sentry.react"
```

Comment (krystofwoldrich) [inline android/build.gradle:17]:
```
        namespace = "io.sentry.react"
```

**Comment by krystofwoldrich** [comment]
Based on [this comment from the RN proposal](https://github.com/react-native-community/discussions-and-proposals/issues/671#issuecomment-1597386954) we will have to keep the `package` in the `AndroidManifest` to ensure RN CLI in older versions recognize the sentry package.

File: android/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -1,5 +1,4 @@
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="io.sentry.react">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (krystofwoldrich) [inline android/src/main/AndroidManifest.xml:1]:
```suggestion
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="io.sentry.react">
```

**Review by krystofwoldrich — APPROVED** [review]
[Successful CI](https://github.com/getsentry/sentry-react-native/pull/3133/commits/d1b9f08991e23898334cd448a604c5007c8deff2)

Thank you @peterlazar1993 for the PR.

**Comment by anurag-hugo** [comment]
hey @krystofwoldrich. We have upgraded our react-native version to 0.72.1 and Sentry started throwing an error during the Android app build because of the missing namespace. When is the next release planned?
Thank you.

**Comment by krystofwoldrich** [comment]
@anurag-hugo The next release will be in the upcoming days.

**Pull request merged**